src/torchrunx/launcher.py

```diff
@@ -97,6 +98,7 @@ class Launcher:
     )
     env_file: str | os.PathLike | None = None
     timeout: int = 600
+    auto: bool = False
```

Comment:
I would consider change the order and defaults to
```python
auto: bool = False
hostnames: list[str] | None =["localhost"]
workers_per_host: int | list[int] | None = 1
```

Then, if the user wants `auto` mode: they would do `Launcher(auto=True, hostnames=None, workers_per_host=None, ...)`